### PR TITLE
examples: fix style w.r.t env var usage, const placement

### PR DESCRIPTION
**Pull request by cpu**
Resolving [feedback from djc](https://github.com/rustls/rustls/pull/1715#discussion_r1440697185) on https://github.com/rustls/rustls/pull/1715

### CODE REVIEW

File: examples/src/bin/simpleserver.rs [hunk 1]
```diff
@@ -9,27 +9,30 @@
...
-
 fn main() -> Result<(), Box<dyn StdError>> {
-    let certs = rustls_pemfile::certs(&mut BufReader::new(&mut File::open(CERTFILE)?))
+    let mut args = env::args();
+    let cert_file = match args.next() {
+        Some(certfile) => certfile,
+        None => "localhost.pem".into(),
+    };
+
+    let private_key_file = match args.next() {
```
Comment (ctz) [inline examples/src/bin/simpleserver.rs:26]: Not sure it really makes sense to allow one argument (which will, from my reading, be the certificate file name with the private key taken from the "localhost-key.pem" default). I'd be in favour of dropping the defaults altogether, and just requiring two arguments.

Comment (cpu) [reply examples/src/bin/simpleserver.rs]: Good point. I've simplified.

**Review by ctz — APPROVED** [review]

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Eventually I think we should use clap in derive mode for all the examples (especially the ones currently using docopt), but that can probably wait.

**Comment by cpu** [comment]
> Eventually I think we should use clap in derive mode for all the examples (especially the ones currently using docopt)

I agree for the ones using docopt, but I think the "simple" examples benefit from just being dead simple Rust with as few dependencies above/beyond Rustls as possible.

**Pull request merged**